client.go

```diff
@@ -1307,13 +1307,21 @@ func (l *link) muxDetach() {
 		Closed: true,
 		Error:  detachError,
 	}
-	select {
-	case l.session.tx <- fr:
-	case <-l.session.done:
-		if l.err == nil {
-			l.err = l.session.err
+
+Loop:
+	for {
+		select {
+		case l.session.tx <- fr:
+			// after sending the detach frame, break the read loop
+			break Loop
+		case _ = <-l.rx:
+			// discard link frames since we are sending a detach
```

Comment:
We'll need to handle the case where a detach is received here.

Copying the `if fr, ok := fr.(*performDetach); ok && fr.Closed {` check from below and setting `l.detachReceived = true` should do the trick.